app/src/main/java/org/mozilla/firefox/vpn/main/vpn/GuardianVpnService.kt

```diff
@@ -18,9 +24,30 @@ class GuardianVpnService : WireGuardVpnService() {
     }
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        val turnOn = {
+            startForeground(NotificationUtil.DEFAULT_NOTIFICATION_ID, NotificationUtil.createBaseBuilder(this).build())
+        }
         when (intent?.getStringExtra(EXTRA_COMMAND)) {
-            COMMAND_TURN_ON -> startForeground(NotificationUtil.DEFAULT_NOTIFICATION_ID, NotificationUtil.createBaseBuilder(this).build())
-            COMMAND_TURN_OFF -> stopForeground(false)
+            COMMAND_TURN_OFF -> {
+                Log.d(logTag, "Handling 'turn off' command")
+                // Service being turned off manually by the user.
+                stopForeground(false)
```

Comment:
I am missing context: Would we want to `stopSelf()` here too? Without being in the foreground that may happen eventually anyways, I assume?

Reply:
I think the intention here is to make sure `return super.onStartCommand` actually runs - wireguard will do its own handling of "command_turn_off", turning itself off, notifying listeners (e.g. we listen for its "down" events to reset an uptime counter).

I'm not sure if calling `stopSelf` explicitly would interfere here, I suspect it may.